### DISCUSSION TITLE
ExplicitExpansion on specific columns

### DISCUSSION THREAD

**Discussion by KryptoBeard** [AutoMapper/AutoMapper.Extensions.OData#175] [Q&A]
Hello. I have the model below. Status and Type are both Objects that I have set to expand automatically in the EDM.

I don't want addresses, contacts, or hours unless expanded. Which is working great. The issue is, it is no longer bringing back the Status or Type objects and it was before. 

```csharp
CreateMap<MyModel, MyModelResource>()
         .ForMember(f => f.Type, o => o.MapFrom(f => f.Type))
          .ForMember(f => f.Status, o => o.MapFrom(f => f.Status))
          .ForMember(f=>f.Addresses,o=>o.MapFrom(f=>f.Addresses))
          .ForMember(f=>f.Hours,o=>o.MapFrom(f=>f.Hours))
          .ForMember(f=>f.Contacts,o=>o.MapFrom(f=>f.Contacts))
          .ForMember(f=>f.Addresses,o=>o.ExplicitExpansion())
          .ForMember(f=>f.Hours,o=>o.ExplicitExpansion())
          .ForMember(f=>f.Contacts,o=>o.ExplicitExpansion())
```

```csharp
 var results = await _myService
      .GetModels()
      .AsNoTracking()
      .GetAsync(_map,options,querySettings);
```

```csharp
 builder.EntitySet<MyResource>("MyResource")
      .EntityType
      .Expand(SelectExpandType.Automatic, "Type", "Status");
```

Works fine when removing explicit, but trying to remove the query for contacts, addresses and hours. Which this does, but it's not brining back Type or Status as I would think it would.

Not bringing in Type, or Status. Which was not set to be explicit.

Thanks!

**Comment by BlaiseD** [comment]
I would look at the code [here](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/b44c99cfa0afdb2a1ab5a7f4d310ea0ae3f42d90/AutoMapper.AspNetCore.OData.EFCore/LinqExtensions.cs#L467) or [here](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/b44c99cfa0afdb2a1ab5a7f4d310ea0ae3f42d90/AutoMapper.AspNetCore.OData.EFCore/LinqExtensions.cs#L501) for `GetQueryAsync` to see it they are among the expansions.  Consider inspecting the code for the [release](https://github.com/AutoMapper/AutoMapper.Extensions.OData/releases) where it worked as expected.

**Comment by Gustavo-Botan** [comment]
Try using **ForPath** instead of **ForMember** for the Status or Type fields.
Customize configuration for a path inside the destination object.
